### DISCUSSION TITLE
What’s the use of bouncer ?

### DISCUSSION THREAD

**Discussion by HarshCodes-07** [adonisjs/core#4425] [Help]
What does it solve that can't be solved myself using utility classes and placing logic separate and encapsulating well. 
Just want to know the logic behind it.

**Comment by GabrielNBDS** [comment]
By that logic you also wouldn't need auth. Bouncer just give some little utils to add a patter to authorization across apps. Probably what it gives that would be harder to do by yourself are the @can and @cannot tags.
